src/main/java/com/opentable/db/postgres/embedded/EmbeddedPostgres.java

```diff
@@ -637,16 +645,15 @@ private static String getArchitecture()
      * Unpack archive compressed by tar with xz compression. By default system tar is used (faster). If not found, then the
      * java implementation takes place.
      *
-     * @param tbzPath The archive path.
+     * @param stream    An input stream with the postgres binary.
      * @param targetDir The directory to extract the content to.
      */
-    private static void extractTxz(String tbzPath, String targetDir) throws IOException
-    {
+    private static void extractTxz(InputStream stream, String targetDir) throws IOException {
         try (
-                InputStream in = Files.newInputStream(Paths.get(tbzPath));
-                XZInputStream xzIn = new XZInputStream(in);
+                XZInputStream xzIn = new XZInputStream(stream);
                 TarArchiveInputStream tarIn = new TarArchiveInputStream(xzIn)
         ) {
+            Phaser phaser = new Phaser(1);
```

Comment:
I'm not super familiar with Phasers. Any reason you'd use this instead of a CountdownLatch?

Reply:
I use a phaser because the number of parties registered to synchronize may vary over time. To use CountDownLatch, I would have to know the number of all files before starting the processing. The phaser allows to register/deregister tasks at any time.